Sources/Aardvark/Bug Reporting/LogStoreAttachmentGenerator.swift

```diff
@@ -0,0 +1,142 @@
+            } else {
+                screenshotAttachment = nil
+            }
+
+            let logsAttachment = attachment(
+                for: logMessages,
+                using: messageFormatter,
+                logStoreName: logStore.name
+            )
+
+            completionQueue.async {
+                completion(
+                    .init(
+                        logMessagesAttachment: logsAttachment,
+                        latestScreenshotAttachment: screenshotAttachment
+                    )
+                )
+            }
+        }
+    }
+
+    /// Generates an attachment containing the latest screenshot in the provided log messages.
+    ///
+    /// - parameter logMessages: The log messages through which to search for the screenshot.
+    /// - parameter logStoreName: The name of the log store from which the logs were collected.
+    @objc(attachmentForLatestScreenshotInLogMessages:logStoreName:)
+    public static func attachmentForLatestScreenshot(
+        in logMessages: [ARKLogMessage],
+        logStoreName: String?
+    ) -> ARKBugReportAttachment? {
+        guard
+            let screenshotMessage = logMessages.reversed().first(where: { $0.type == .screenshot }),
+            let imageData = screenshotMessage.image?.pngData()
+        else {
+            return nil
+        }
+
+        return ARKBugReportAttachment(
+            fileName: screenshotFileName(for: logStoreName),
+            data: imageData,
+            dataMIMEType: "image/png"
+        )
+    }
+
+    /// Generates an attachment containing the log messages formatted using the specified formatter.
+    ///
+    /// - parameter logMessages: The log messages to be included in the attachment.
+    /// - parameter logFormatter: The formatter with which to format the log messages.
+    /// - parameter logStoreName: The name of the log store from which the logs were collected.
+    @objc(attachmentForLogMessages:usingLogFormatter:logStoreName:)
+    public static func attachment(
+        for logMessages: [ARKLogMessage],
+        using logFormatter: ARKLogFormatter,
+        logStoreName: String?
+    ) -> ARKBugReportAttachment {
+        let formattedLogData = logMessages
+            .map(logFormatter.formattedLogMessage(_:))
+            .joined(separator: "\n")
+            .data(using: .utf8)!
+
+        let fileName = logsFileName(for: logStoreName, fileType: "txt")
+
+        return ARKBugReportAttachment(
+            fileName: fileName,
+            data: formattedLogData,
+            dataMIMEType: "text/plain"
+        )
+    }
+
+    // MARK: - Private Static Methods
+
+    private static func screenshotFileName(for logStoreName: String?) -> String {
+        var fileName = NSLocalizedString("screenshot", comment: "File name of a screenshot")
+        fileName = (fileName as NSString).appendingPathExtension("png") ?? fileName
```

Comment:
`appendingPathExtension` can fail?? Weird...

Reply:
I think converting to a URL would just move the nullability issue to a new spot, rather than keeping it non-null the whole time 😞

Reply:
@dfed As usual, you are correct. 🤦 Updated in d9ffbb5.